internal/store/store.go

```diff
@@ -181,6 +181,26 @@ func (bs *BlockStore) LoadBlockByHash(hash []byte) *types.Block {
 	return bs.LoadBlock(height)
 }
 
+// LoadBlockMetaByHash returns the blockmeta who's header corresponds to the given
+// hash. If none is found, returns nil.
+func (bs *BlockStore) LoadBlockMetaByHash(hash []byte) *types.BlockMeta {
+	bz, err := bs.db.Get(blockHashKey(hash))
+	if err != nil {
+		panic(err)
```

Comment:
I see this pattern is pre-existing, but I'm sad that we do this. 🙁

Reply:
Yeah I'd love to add error handling (propagate it up to the caller) to the blockstore (we do this actually in the state store). Maybe I can create an issue so we don't forget

Reply:
Opened this: https://github.com/tendermint/tendermint/issues/7366